src/MessageHandler/DatasetSubmissionFilerHandler.php

```diff
@@ -166,17 +187,24 @@ private function processFile(File $file): void
             $file->setStatus(File::FILE_ERROR);
         }
 
-
         // File Hashing
         $hashFile = new HashFile($fileId);
         $this->messageBus->dispatch($hashFile);
 
         // File virus Scan
-        // TODO implement method
-        /*
-        $scanFile = new VirusScan($fileId);
-        $this->messageBus->dispatch($scanFile);
-        */
+        if ($file instanceof File) {
+            $filename = $this->datastoreDirectory . '/' . $file->getFilePath();
```

Comment:
We don't need to do this, we can inject Datastore util class, and get the FileStream. Use getFile() to get the stream